### PR TITLE
Allow a deprecated attribute to access its replacement

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

Deprecating a public attribute using `astropy.utils.decorators.deprecated_attribute()` currently requires setting up a corresponding private attribute even if it is being replaced by a new public attribute. This pull request simplifies renaming public attributes by allowing the deprecated attribute access its replacement.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/utils/tests/test_decorators.py [hunk 1]
```diff
@@ -57,17 +56,18 @@ def set_private(self):
 
     with pytest.warns(NewDeprecationWarning, match="The bar attribute is "
                       "deprecated and may be removed in a future version.") as w:
-        dummy.bar
+        dummy.bar == 4242
```
Comment (pllim) [inline astropy/utils/tests/test_decorators.py:59]: Do you mean to use `==` instead of `=`?

**Review by pllim — COMMENTED** [review]
I vaguely remember @bsipocz being interested in these functionalities, so pinging her in case she's still interested to review.

File: astropy/utils/tests/test_decorators.py (hunk 1 above)
Comment (eerovaher) [reply astropy/utils/tests/test_decorators.py]: The new commits update the tests properly.

**Comment by pllim** [comment]
Code LGTM. Can you please add a change log? Thanks!

**Comment by pllim** [comment]
I flip flop between bug fix and new feature. If you think this is really a bug fix, then we also have to backport it.

**Comment by eerovaher** [comment]
This can indeed be considered a bugfix, so I've edited the labels and milestone accordingly.

**Review by pllim — APPROVED** [review]
Thanks!

**Comment by pllim** [comment]
Tsk tsk, numpy changed things up again in dev. Failure is unrelated.

**Pull request merged**